Drop unsampled transactions when connected to APM Server 8.0+ (#554)
elastic/apm: specs/terminology.md

## Sampling

To reduce processing and storage overhead, transactions may be "sampled".
Non-sampled transactions and spans will not be stored in Elasticsearch.

> In versions prior to 8.0, non-sampled transactions did get stored but sampling limits the amount of data that is captured for transactions:
> non-sampled transactions will not record context, and related spans will not be captured.

Reviewer: ```suggestion

In versions prior to 8.0, non-sampled transactions were stored, but no context was recorded and no related spans were captured.
```
